### PR TITLE
LPS-89497 Cover the configuration doesn't have this property

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang,  if the last commit is good for you, please help merge the last two commits to one commit.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/upgrade/test/ConfigurationUpgradeStepFactoryTest.java [hunk 1]
```diff
@@ -130,12 +130,19 @@ private void _testUpgradeConfig(
...
-					configuration,
-					MapUtil.singletonDictionary(
-						"felix.fileinstall.filename", uri.toString()));
+					ConfigurationTestUtil.saveConfiguration(
+						configuration,
+						MapUtil.singletonDictionary(
+							"felix.fileinstall.filename", uri.toString()));
+				}
+				else {
+					ConfigurationTestUtil.saveConfiguration(
```
Comment (dantewang) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/upgrade/test/ConfigurationUpgradeStepFactoryTest.java]: @yuhai Why do you put an irrelevant property into the dictionary?

Comment (yuhai) [reply modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/upgrade/test/ConfigurationUpgradeStepFactoryTest.java]: @dantewang, this is only for the api using, the api doesn't allow the key is empty. I wanted to cover the property is not null when dataExist and configFileExist.  And cover the property is null when dataExist and !configFileExist.

**Comment by dantewang** [comment]
@yuhai What about not adding anything to the dictionary?

Yu Hai <notifications@github.com> 于 2020年3月3日周二 19:47写道：

**Comment by yuhai** [comment]
@dantewang, if using MapUtil.singletonDictionary("", ""), it will throw the exception

```
java.lang.IllegalArgumentException: Key [] must not be an empty string
        at org.apache.felix.cm.impl.CaseInsensitiveDictionary.checkKey(CaseInsensitiveDictionary.java:269)
        at org.apache.felix.cm.impl.CaseInsensitiveDictionary.<init>(CaseInsensitiveDictionary.java:73)
        at org.apache.felix.cm.impl.ConfigurationImpl.update(ConfigurationImpl.java:452)
        at org.apache.felix.cm.impl.ConfigurationAdapter.update(ConfigurationAdapter.java:145)
        at com.liferay.portal.configuration.test.util.ConfigurationTestUtil._updateProperties(ConfigurationTestUtil.java:178)
        at com.liferay.portal.configuration.test.util.ConfigurationTestUtil.saveConfiguration(ConfigurationTestUtil.java:77)
```

**Comment by dantewang** [comment]
@yuhai I mean don't even operate on the dictionary...

Yu Hai <notifications@github.com> 于 2020年3月3日周二 20:22写道：

**Comment by dantewang** [comment]
Fetched

**Comment by dantewang** [comment]
Pull request submitted to https://github.com/yuhai/liferay-portal/pull/210. [See changes here.](https://github.com/dantewang/liferay-portal/compare/cf3a978fcd04~2...cf3a978fcd04)

**Pull request closed without merging**